### PR TITLE
Feature/tree image scrubber

### PR DESCRIPTION
**Pull request by misterinterrupt**

### CODE REVIEW

**Comment by misterinterrupt** [comment]
I added the 'approved' field to the tree model in tree.json

File: server/common/models/tree.json [hunk 1]
```diff
@@ -335,6 +335,21 @@
         "dataScale": null,
         "nullable": "YES"
       }
+    },
+    "approved": {
```
Comment (misterinterrupt) [inline server/common/models/tree.json:339]: heres the new tree model field

File: client/src/api/treeTrackerApi.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  const query = `${baseUrl}/trees/${id}`;
+  return fetch(query, {
+    method: "PATCH",
+    headers: { "content-type": "application/json" },
+    body: JSON.stringify({ approved: true })
+  })
+    .then(handleResponse)
+    .catch(handleError);
+}
+
```
Comment (misterinterrupt) [inline client/src/api/treeTrackerApi.js:27]: the approve api call

File: client/src/api/treeTrackerApi.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+export function rejectTreeImage(id) {
+  const query = `${baseUrl}/trees/${id}`;
+  return fetch(query, {
+    method: "PATCH",
+    headers: { "content-type": "application/json" },
+    body: JSON.stringify({ active: false })
+  })
+    .then(handleResponse)
+    .catch(handleError);
+}
```
Comment (misterinterrupt) [inline client/src/api/treeTrackerApi.js:37]: the reject api call

File: client/src/api/apiUtils.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  }
+  throw new Error("Network response was not ok.");
+}
+
+// we should call an error logging service, but
+export function handleError(error) {
+  // eslint-disable-next-line no-console
+  console.error("API call failed. " + error);
+  throw error;
+}
```
Comment (misterinterrupt) [inline client/src/api/apiUtils.js:17]: this file has some simple api interaction functions

File: client/src/App.js [hunk 5]
```diff
@@ -1,40 +1,42 @@
...
   requestTreeCount: id => dispatch.trees.requestTreeCount(),
   requestTrees: id => dispatch.trees.requestTrees()
-})
+});
 
-export default connect(mapState, mapDispatch)(App)
+export default connect(
+  mapState,
+  mapDispatch
+)(App);
```
Comment (misterinterrupt) [inline client/src/App.js:42]: linting fixes only

File: client/src/components/AppDrawer.js [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+
+AppDrawer.propTypes = {};
+
+export default compose(
+  withStyles(styles, { withTheme: true, name: "AppDrawer" }),
+  connect(
+    mapState,
+    mapDispatch
+  )
+)(AppDrawer);
```
Comment (misterinterrupt) [inline client/src/components/AppDrawer.js:165]: this file was moved

File: client/src/components/AppFrame.js [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+      theme,
+      toggleAppDrawer,
+      appDrawer,
+      currentView
+    } = this.props;
+    let tabContents;
+    if (currentView === "trees") {
+      tabContents = <Trees />;
+    } else if (currentView === "imageScrubber") {
+      tabContents = <TreeImageScrubber />;
```
Comment (misterinterrupt) [inline client/src/components/AppFrame.js:97]: this is where I replaced the old scrubber with the new treeImageScrubber

File: client/src/components/AppFrame.js [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+  theme: PropTypes.object.isRequired
+};
+
+export default compose(
+  withStyles(styles, { withTheme: true, name: "AppFrame" }),
+  connect(
+    mapState,
+    mapDispatch
+  )
+)(AppFrame);
```
Comment (misterinterrupt) [inline client/src/components/AppFrame.js:166]: this file was moved

File: client/src/components/Icon.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+          className={
+            props.active ? props.classes.active : props.classes.default
+          }
+          d={iconPaths[props.icon]}
+        />
+      </svg>
+    );
+  }
+});
+export default withStyles(styles)(Icon);
```
Comment (misterinterrupt) [inline client/src/components/Icon.js:28]: moved

File: client/src/components/TreeImageScrubber.js [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+          );
+        }
+      })}
+    </section>
+  );
+}
+
+export default compose(
+  withStyles(styles, { withTheme: true, name: "ImageScrubber" })
+)(TreeImageScrubber);
```
Comment (misterinterrupt) [inline client/src/components/TreeImageScrubber.js:157]: new TreeImageScrubber that uses react hooks and local state

File: client/src/components/TreeTable.js [hunk 11]
```diff
@@ -0,0 +1,251 @@
...
+  getTreeAsync: id => dispatch.trees.getTreeAsync(id)
+});
+
+export default compose(
+  withStyles(styles, { withTheme: true, name: "TreeTable" }),
+  connect(
+    mapState,
+    mapDispatch
+  )
+)(TreeTable);
```
Comment (misterinterrupt) [inline client/src/components/TreeTable.js:251]: moved

File: client/src/components/Trees.js [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+const mapDispatch = dispatch => {
+  return {};
+};
+
+Trees.propTypes = {};
+
+export default connect(
+  mapState,
+  mapDispatch
+)(Trees);
```
Comment (misterinterrupt) [inline client/src/components/Trees.js:39]: moved

File: client/src/models/imageScrubber.js [hunk 13]
```diff
@@ -1,113 +1,107 @@
...
-        .catch(err => console.error(`ERROR: FAILED TO RETRIEVE STATUS ${err}`))
-    },
+        .catch(err => console.error(`ERROR: FAILED TO RETRIEVE STATUS ${err}`));
+    }
   }
-}
+};
 
-export default imageScrubber
+export default imageScrubber;
```
Comment (misterinterrupt) [inline client/src/models/imageScrubber.js:107]: not sure what happened with the merge, but this was moved and is no longer being used. we can remove it later

Comment (misterinterrupt) [reply client/src/models/imageScrubber.js:107]: this preview is not so useful, if interested, see the full changes

**Comment by ZavenArra** [comment]
Yaaaay!

**Pull request merged**